Sources/OpenAPIRuntime/Interface/HTTPBody.swift

```diff
@@ -0,0 +1,535 @@
+                    )
+                }
+                return transformedElement
+            }
+        case .unknown:
+            validatedTransform = transform
+        }
+        return HTTPBody(
+            sequence: map(validatedTransform),
+            length: length,
+            iterationBehavior: iterationBehavior
+        )
+    }
+}
+
+// MARK: - Consumption utils
+
+extension HTTPBody {
+
+    /// An error thrown by the `collect` function when the body contains more
+    /// than the maximum allowed number of bytes.
+    private struct TooManyBytesError: Error, CustomStringConvertible, LocalizedError {
+        let maxBytes: Int
+
+        var description: String {
+            "OpenAPIRuntime.HTTPBody contains more than the maximum allowed \(maxBytes) bytes."
+        }
+
+        var errorDescription: String? {
+            description
+        }
+    }
+
+    /// An error thrown by the `collect` function when another iteration of
+    /// the body is not allowed.
+    private struct TooManyIterationsError: Error, CustomStringConvertible, LocalizedError {
+
+        var description: String {
+            "OpenAPIRuntime.HTTPBody attempted to create a second iterator, but the underlying sequence is only safe to be iterated once."
+        }
+
+        var errorDescription: String? {
+            description
+        }
+    }
+
+    /// Accumulates the full body in-memory into a single buffer
+    /// up to `maxBytes` and returns it.
+    /// - Parameters:
+    ///   - maxBytes: The maximum number of bytes this method is allowed
+    ///     to accumulate in memory before it throws an error.
+    /// - Throws: `TooManyBytesError` if the the sequence contains more
+    ///   than `maxBytes`.
+    public func collect(upTo maxBytes: Int) async throws -> DataType {
+
+        // As a courtesy, check if another iteration is allowed, and throw
+        // an error instead of fatalError here if the user is trying to
+        // iterate a sequence for the second time, if it's only safe to be
+        // iterated once.
+        if iterationBehavior == .single {
+            try {
+                lock.lock()
+                defer {
+                    lock.unlock()
+                }
+                guard !locked_iteratorCreated else {
+                    throw TooManyIterationsError()
+                }
+            }()
+        }
+
+        var buffer = DataType.init()
+        for try await chunk in self {
+            guard buffer.count + chunk.count <= maxBytes else {
+                throw TooManyBytesError(maxBytes: maxBytes)
+            }
+            buffer.append(contentsOf: chunk)
+        }
+        return buffer
+    }
+}
+
+// MARK: - String-based bodies
+
+extension StringProtocol {
+    fileprivate var asBodyChunk: HTTPBody.DataType {
```

Comment:
`@inlinable` or otherwise quite slow as this is a generic over `StringProtocol`

Reply:
This is an internal symbol, I thought the compiler doesn't need inlinable annotations for internal symbols, only public ones? (It should be inlining these even without the annotation I thought.)

Reply:
You call this in a public generic function (`HTTPBody.init(data: some StringProtocol, ...)`) which forwards the generic type to this function call. Therefore it needs to be @inlinable or it can't be specialised.